src/olympia/lib/crypto/tasks.py

```diff
@@ -91,49 +60,58 @@ def get_new_version_number(version):
 def sign_addons(addon_ids, force=False, **kw):
     """Used to sign all the versions of an addon.
 
-    This is used in the 'process_addons --task sign_addons' management
-    command.
+    This is used in the 'process_addons --task resign_addons_for_cose'
+    management command.
 
     It also bumps the version number of the file and the Version, so the
     Firefox extension update mechanism picks this new signed version and
     installs it.
     """
     log.info(u'[{0}] Signing addons.'.format(len(addon_ids)))
 
-    reasons = {
-        'default': [MAIL_SUBJECT, MAIL_MESSAGE],
-        'expiry': [MAIL_EXPIRY_SUBJECT, MAIL_EXPIRY_MESSAGE]
-    }
-    mail_subject, mail_message = reasons[kw.get('reason', 'default')]
+    mail_subject, mail_message = MAIL_COSE_SUBJECT, MAIL_COSE_MESSAGE
+
+    # query everything except for search-plugins as they're generally
+    # not signed
+    current_versions = (
+        Addon.objects
+        .filter(id__in=addon_ids)
+        .exclude(type=amo.ADDON_SEARCH)
```

Comment:
Shouldn't this be done earlier, in `process_addons` task declaration ?

Reply:
Actually, I just re-read the `process-addon` command and I think you're right, all that should be moving there. I'll fix that.